modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java

```diff
@@ -135,6 +135,11 @@
  *         }
  *     }
  * </pre>
+ * <strong>Warning:</strong>
+ * Serialized objects of this class will not be compatible with
+ * future Swing or JavaFX releases. The current serialization support is
+ * appropriate for short term storage or RMI between applications running
+ * the same version of Swing and the same version of JavaFX.
```

Comment:
Shouldn;t we need to add
@SuppressWarnings("serial") // Same-version serialization only

which I see in almost all classes in Swing which has the above serialization warning...

Reply:
This PR is addressing doclint warnings, not javac lint warnings. javac warnings can be addressed in a follow-up bug.

Reply:
FYI: after merging this and #1384 together all I see in the build log are 4 native warnings (on macOS):

```
/Users/angorya/Projects/jfx-1/jfx/rt/modules/javafx.graphics/src/main/native-prism-es2/macosx/MacOSXWindowSystemInterface.m:93:21: warning: 'lockFocusIfCanDraw' is deprecated: first deprecated in macOS 10.14 - To draw, subclass NSView and implement -drawRect:; AppKit's automatic deferred display mechanism will call -drawRect: as necessary to display the view. [-Wdeprecated-declarations]
        if ([nsView lockFocusIfCanDraw] == NO) {
                    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSView.h:190:1: note: 'lockFocusIfCanDraw' has been explicitly marked deprecated here
- (BOOL)lockFocusIfCanDraw API_DEPRECATED("To draw, subclass NSView and implement -drawRect:; AppKit's automatic deferred display mechanism will call -drawRect: as necessary to display the view.", macos(10.0,10.14));
^
/Users/angorya/Projects/jfx-1/jfx/rt/modules/javafx.graphics/src/main/native-prism-es2/macosx/MacOSXWindowSystemInterface.m:98:25: warning: 'unlockFocus' is deprecated: first deprecated in macOS 10.14 - To draw, subclass NSView and implement -drawRect:; AppKit's automatic deferred display mechanism will call -drawRect: as necessary to display the view. [-Wdeprecated-declarations]
                [nsView unlockFocus];
                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSView.h:189:1: note: 'unlockFocus' has been explicitly marked deprecated here
- (void)unlockFocus API_DEPRECATED("To draw, subclass NSView and implement -drawRect:; AppKit's automatic deferred display mechanism will call -drawRect: as necessary to display the view.", macos(10.0,10.14));
^
/Users/angorya/Projects/jfx-1/jfx/rt/modules/javafx.graphics/src/main/native-prism-es2/macosx/MacOSXWindowSystemInterface.m:120:24: warning: 'setView:' is deprecated: first deprecated in macOS 10.14 - Use NSOpenGLView to provide OpenGL content in a Cocoa app. [-Wdeprecated-declarations]
            [nsContext setView : nsView];
                       ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSOpenGL.h:193:36: note: property 'view' is declared deprecated here
@property (nullable, weak) NSView *view NS_SWIFT_UI_ACTOR API_DEPRECATED("", macos(10.0,10.14));
                                   ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSOpenGL.h:194:1: note: 'setView:' has been explicitly marked deprecated here
- (void)setView:(nullable NSView *)view NS_SWIFT_UI_ACTOR API_DEPRECATED("Use NSOpenGLView to provide OpenGL content in a Cocoa app.", macos(10.0,10.14));
^
/Users/angorya/Projects/jfx-1/jfx/rt/modules/javafx.graphics/src/main/native-prism-es2/macosx/MacOSXWindowSystemInterface.m:121:21: warning: 'unlockFocus' is deprecated: first deprecated in macOS 10.14 - To draw, subclass NSView and implement -drawRect:; AppKit's automatic deferred display mechanism will call -drawRect: as necessary to display the view. [-Wdeprecated-declarations]
            [nsView unlockFocus];
                    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSView.h:189:1: note: 'unlockFocus' has been explicitly marked deprecated here
- (void)unlockFocus API_DEPRECATED("To draw, subclass NSView and implement -drawRect:; AppKit's automatic deferred display mechanism will call -drawRect: as necessary to display the view.", macos(10.0,10.14));
^
4 warnings generated.
```


as well as a number of these messages:

```
-do-compile:
    [mkdir] Created dir: /Users/angorya/Projects/jfx-1/jfx/rt/apps/toys/PickTest3D/build/empty
    [mkdir] Created dir: /Users/angorya/Projects/jfx-1/jfx/rt/apps/toys/PickTest3D/build/generated-sources/ap-source-output
    [javac] Compiling 6 source files to /Users/angorya/Projects/jfx-1/jfx/rt/apps/toys/PickTest3D/build/classes
    [javac] warning: [options] system modules path not set in conjunction with -source 9
    [javac] Note: /Users/angorya/Projects/jfx-1/jfx/rt/apps/toys/PickTest3D/src/picktest/PickTest3D.java uses unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
    [javac] 1 warning
```

Reply:
That's because we don't enable javac lint warnings by default. I am getting ready to file a few JBS issues related to this. Stay tuned.

In the mean time, if you want to see how many warnings we do have, try running with `gradle -PLINT=all`. Or rather, you'll see how many modules hit the 100 warnings limit. :)

Reply:
So, that
@SuppressWarnings("serial") // Same-version serialization only
would be addressed as javac lint warning JBS?

Just thinking out aloud, I dont see this @SuppressWarnings("doclint:missing") in our java.desktop classes? what's the difference there?

Reply:
> So, that @SuppressWarnings("serial") // Same-version serialization only would be addressed as javac lint warning JBS?

Yes. I filed [JDK-8327170](https://bugs.openjdk.org/browse/JDK-8327170) as an umbrella task to address javac lint warnings, and will file new linked blocking bugs to address individual warnings.

> Just thinking out aloud, I dont see this @SuppressWarnings("doclint:missing") in our java.desktop classes? what's the difference there?

We wanted to silence the warnings (so we can turn on `-Werror` once the last warning is gone), and didn't see any real benefit in actually documenting the private fields. We could revisit that later if there was a reason, but I suspect there isn't.